src/app/public/modules/href/href.directive.ts

```diff
@@ -0,0 +1,222 @@
+import { HttpParams } from '@angular/common/http';
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  Input,
+  Optional,
+  Renderer2
+} from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+import { SkyAppConfig, SkyAppRuntimeConfigParamsProvider } from '@skyux/config';
+import { Subject } from 'rxjs';
+
+import { SkyHrefQueryParams } from './href-query-params';
+import { SkyHrefResolverService } from './href-resolver.service';
+
+import { AppRoute } from './types/app-route';
+
+type HrefChanges = { href: string, styleDisplay: string };
+
+@Directive({
+  selector: '[skyHref]'
+})
+export class SkyHrefDirective {
+
+  @Input()
+  public skipLocationChange = false;
+
+  @Input()
+  public replaceUrl = false;
+
+  @Input()
+  public state?: {[k: string]: any};
+
+  @Input()
+  public set queryParams(value: SkyHrefQueryParams) {
+    this._queryParams = value;
+    this._changes.next(this.getChanges());
+  }
+
+  @Input()
+  public set skyHref(skyHref: string) {
+    this._skyHref = skyHref;
+    this.checkRouteAccess();
+  }
+
+  @Input()
+  public set skyHrefElse(value: 'hide' | 'unlink') {
+    this._skyHrefElse = value;
+    this._changes.next(this.getChanges());
+  }
+
+  private _changes: Subject<HrefChanges>;
```

Comment:
What's the purpose of having a `_changes` `Subject`? Since changes are only ever pushed to it internal to this class, could you not accomplish the same thing with a simpler method like `applyChanges()`? Then you don't have to worry about unsubscribing when the directive is destroyed.